src/main/java/com/example/ForexApp_v1/controller/TransactionController.java

```diff
@@ -4,77 +4,92 @@
 import java.util.List;
 
+
 @Controller
+@Validated
 public class TransactionController {
     private final TransacService transacService;
     private List<TransacDTO> transacDTOList = new ArrayList<>();
     public TransactionController(TransacService transacService) {
         this.transacService = transacService;
     }
-
-    //adnotacja aby sprawdzać czy dane nie są puste bądź nullem ---np noEmpty
     @GetMapping("/singleTransaction")
     public String showToSingleTransaction(Model model) {
         try {
             model.addAttribute("transacDTO", new TransacDTO());
             return "singleTransaction";
         }catch (Exception e){
             e.printStackTrace();
-            return "index"; // wstawić stronę że błąd
+            return "singleTransaction";
         }
     }
+
     @PostMapping("/singleTransaction")
-    public String addToSingleTransaction(@ModelAttribute("transacDTO") TransacDTO transacDTO, Model model){
-      try {
-          Transac transac = transacService.addTransaction(transacDTO);
+        public String addToSingleTransaction(final @Valid @ModelAttribute("transacDTO") TransacDTO transacDTO, final Model model, final BindingResult bindingResult){
+        if (bindingResult.hasErrors()) {
+            return "singleTransaction";
+        }
+        try {
+          TransacDTO transac = transacService.addTransaction(transacDTO);
           model.addAttribute("transac", transac);
           return "singleTransaction";
-      }catch (Exception e){
+      }catch (ConstraintViolationException cve){
+            cve.printStackTrace();
+            return "singleTransaction";
+        }
+        catch (Exception e){
           e.printStackTrace();
-          return "index";  // wstawić stronę że błąd
-      }
+            return "singleTransaction";
+        }
     }
-    @GetMapping("/manyTransactions")
-    public String addToManyTransactions(Model model){
-
-        model.addAttribute("transacDTOList", transacDTOList);
-        model.addAttribute("newTransacDTO", new TransacDTO());
 
-        return "manyTransactions";
+    @GetMapping("/manyTransactions")
+    public String addToManyTransactions(final Model model){
+        try {
+            model.addAttribute("transacDTOList", transacDTOList);
+            model.addAttribute("newTransacDTO", new TransacDTO());
+            return "manyTransactions";
+        }catch (Exception e){
+            return "manyTransactions";
+        }
     }
 
     @PostMapping("/addTransactions")
-    public String showCreateFormTransactions(@ModelAttribute TransacDTO transacDTO){
+    public String showCreateFormTransactions(final @ModelAttribute(name = "transacDTO") TransacDTO transacDTO, final BindingResult bindingResult){
+        if (bindingResult.hasErrors()) {
+            return "manyTransactions";
+        }
         try {
             transacDTOList.add(transacDTO);
-
             return "redirect:/manyTransactions";
         }catch (Exception e){
```

Comment:
Can you make the exception less generic?

Reply:
ok